### ISSUE TITLE
False-positive for NoUnusedImport rule when import ends with 'not'

### ISSUE THREAD

**Issue by fmordashev** [detekt/detekt#5221]
## Expected Behavior
Unused imports should be removed, since my config file includes
```
  NoUnusedImports:
    active: true
    autoCorrect: true
```

## Observed Behavior
The import is not removed if it ends with 'not' text

## Steps to Reproduce
Can be easily tested with any import that ends with 'not'
For example:
```
package my.package.cool.printer

import my.package.cool.matchers.Matcher.not

class Printer() {

  fun printHello() {
     print("hello")
  }
}
```

Initially found for `import org.hamcrest.Matchers.not` from `org.hacrest:hamcrest:2.2`

## Your Environment

* Version of detekt used: tried both 1.19.0 and 1.21.0
* Version of Gradle used (if applicable): 7.4.2
* Gradle scan link (add `--scan` option when running the gradle task): can't provide due to ssl restrictions
* Operating System and version: win10

**Comment by 3flex** [comment]
This is a ktlint rule that detekt wraps for convenience - any false positives or negatives are most likely an issue with the underlying ktlint rule itself and not detekt.

Can you please check if you see the same behaviour when running ktlint directly? If yes, please report this on the ktlint project and close this issue. If not, please report back here and we'll see what we can do.

**Comment by fmordashev** [comment]
Hello, @3flex 

I used "org.jlleitschuh.gradle.ktlint" version "10.3.0" ktlint gradle-plugin and the check indeed didnt detect the unused import from my case.

Is it a valid reproduction or should I use non-plugin ktlint instead?

**Comment by fmordashev** [comment]
created an issue in ktlint rep https://github.com/pinterest/ktlint/issues/1587
